gcc/rust/rust-session-manager.cc

```diff
@@ -249,6 +254,21 @@ Session::handle_option (
   return ret;
 }
 
+bool
+Session::handle_extern_option (std::string &input)
+{
+  auto pos = input.find ('=');
+  if (std::string::npos == pos)
+    {
+      return false;
+    }
+  std::string libname = input.substr (0, pos);
+  std::string path = input.substr (pos + 1);
+
+  extern_crates.insert ({libname, path});
```

Comment:
mh, I'm not sure if we want to keep this as a local hashmap or actually reach into the extern crate importer and add the candidate there. @philberty will have opinions :D

Reply:
I put a hashmap here because I thought this question was kinda out of scope for this PR. Some crates can be renamed with this mechanism, others are proc macros and should be loaded (immediately ? probably not).

Reply:
In fact, rustc even use a [BTreeMap for this kind of thing](https://github.com/rust-lang/rust/blob/master/compiler/rustc_session/src/config.rs#L2181), and advise against any [unsorted containers](https://github.com/rust-lang/rust/blob/master/compiler/rustc_session/src/config.rs#L436).

Reply:
yeah, that's fair. I'm happy with us keeping these crates in the `SessionManager` and then accessing them when loading crates later down the line. IMO it makes sense. and yeah I guess you should switch from `std::unordered_map` to a `std::map` which should be a sorted tree